apps/emqx_bridge_dynamo/src/emqx_bridge_dynamo_connector.erl

```diff
@@ -246,19 +246,26 @@ do_query(
         table := Table,
         templates := Templates
     } = ChannelState,
+    LogMetaData = logger:get_process_metadata(),
+    TraceRenderedFuncContext = #{trace_ctx => LogMetaData, action_id => ChannelId},
+    TraceRenderedFunc = {fun trace_render_result/2, TraceRenderedFuncContext},
     Result =
         case ensuare_dynamo_keys(Query, ChannelState) of
             true ->
                 ecpool:pick_and_do(
                     PoolName,
-                    {emqx_bridge_dynamo_connector_client, query, [Table, QueryTuple, Templates]},
+                    {emqx_bridge_dynamo_connector_client, query, [
+                        Table, QueryTuple, Templates, TraceRenderedFunc
+                    ]},
                     no_handover
                 );
             _ ->
                 {error, missing_filter_or_range_key}
         end,
 
     case Result of
+        {error, {unrecoverable_error, {action_stopped_after_template_rendering, _}}} = Error ->
```

Comment:
maybe use a macro for `{unrecoverable_error, {action_stopped_after_template_rendering, _}}`
e.g. `TRACE_STOP_ACTION(REASON)`

Reply:
I think this is a good idea to avoid typos (will fix).